code/modules/mob/living/living_defense.dm

```diff
@@ -184,13 +184,14 @@
 
 	add_logs(user, src, "grabbed", addition="passively")
 
-	var/obj/item/weapon/grab/G = new /obj/item/weapon/grab(user, src)
+	if(anchored || !user.Adjacent(src))
+		return 0
 	if(buckled)
 		user << "<span class='warning'>You cannot grab [src], \he is buckled in!</span>"
 		return 0
-	if(!G)	//the grab will delete itself in New if src is anchored
-		return 0
-	user.put_in_active_hand(G)
+	var/obj/item/weapon/grab/G = new /obj/item/weapon/grab(user, src)
+	if(!user.put_in_active_hand(G)) //if we can't put the grab in our hand for some reason, we delete it.
+		qdel(G)

```

Comment:
See https://github.com/tgstation/-tg-station/pull/15327/files#diff-0d58be00f177298faed0e648e735056eL27
If we somehow didn't manage to put the grab into our hand, we delete it.